src/compiler/checker.ts

```diff
@@ -5019,7 +5019,8 @@ namespace ts {
                 // If this is a JSDoc construct signature, then skip the first parameter in the
                 // parameter list.  The first parameter represents the return type of the construct
                 // signature.
-                for (let i = isJSConstructSignature ? 1 : 0, n = declaration.parameters.length; i < n; i++) {
+                const n = declaration.parameters.length;
+                for (let i = isJSConstructSignature ? 1 : 0; i < n; i++) {
                     const param = declaration.parameters[i];

```

Comment:
` for (let i = isJSConstructSignature ? 1 : 0; i < declaration.parameters.length; i++) {`